app/main.py

```diff
@@ -1,25 +1,82 @@
 from abc import ABC
+from typing import Any
 
 
 class IntegerRange:
-    pass
+    def __init__(self,
+                 min_amount: int,
+                 max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: Any, name: str) -> Any:
+        self.protected_name = "_" + name
+
+    def __get__(self, instance: Any, value: Any) -> int:
```

Comment:
The `__get__` method has a parameter `value` which is not used. Consider removing it to avoid confusion.